rebac-admin-backend/v1/error.go

```diff
@@ -8,50 +8,53 @@ import (
 	"github.com/canonical/identity-platform-admin-ui/rebac-admin-backend/v1/resources"
 )
 
-// UnauthorizedError represents unauthorized access error.
-type UnauthorizedError struct {
-	message string
-}
-
-func (e *UnauthorizedError) Error() string {
-	return fmt.Sprintf("Unauthorized: %s", e.message)
-}
+// ErrorCode is a code which describes the class of error.
+type ErrorCode string
 
-// NewUnauthorizedError returns a pointer to a new instance of UnauthorizedError with the provided message
-func NewUnauthorizedError(message string) *UnauthorizedError {
-	return &UnauthorizedError{message}
-}
+const (
+	CodeUnauthorized    ErrorCode = "unauthorized"
+	CodeNotFound        ErrorCode = "not found"
+	CodeValidationError ErrorCode = "validation error"
```

Comment:
one thought.. if you weren't dead set on calling it a "validation error" we could just use http.Status* as error codes (which are just ints, if i recall corectly).. then you could also use http.StatusText to map error codes into text.. 
so instead of CodeUnauthorized, you'd use http.StatusUnauthorized, CodeNotFound -> http.StatusNotFound.. and CodeValidationError -> http.StatusBadRequest

Reply:
suggestion: riding Ales' wave I'd like to suggest a change, since the `code` wording seems misleading:
- rename `code` to `status` (so we can also leverage `http.StatusText(code int)` to get the corresponding status string
- we add a `code int` or `statusCode int` to represent the actual http status code

This way we rely on http package for status strings and we only need to map a correct http code like 401 or 400 to build our errors.

The `Error()` function can then leverage the same fields as it does now, just with the renamed `code` into `status` field

Reply:
Thanks for the comment. I applied the changes in the latest commit.